### PR TITLE
feat: implement body/footer parsing

### PR DESCRIPTION
**Pull request by bcoe**
* drops the `body-footer` node type, in favor of a recursive `body` rule.
* implement body + footer parsing.

Screen Shot 2020-12-23 at 7 11 56 PM

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ to keep this in sync with the written specification on conventionalcommits.org.
 /* Any non-newline whitespace: */
 <whitespace>    ::= <ZWNBSP> | <TAB> | <VT> | <FF> | <SP> | <NBSP> | <USP>
 
-<message>       ::= <summary>, <newline>, <bodyfooter>
+<message>       ::= <summary>, <newline>*, <body>
```
Comment (bcoe) [inline README.md:40]: Allow any number of `\n` before the first `body` node.

File: index.js [hunk 2]
```diff
@@ -2,8 +2,8 @@ const Scanner = require('./lib/scanner')
 const { isWhitespace, isNewline, isParens } = require('./lib/type-checks')
 
 /*
- * <message>      ::= <summary> <newline> <body-footer>
- *                 |  <summary>
+ * <message>       ::= <summary>, <newline>?, <body>
```
Comment (byCedric) [inline index.js:5]: shouldn't this be `<newline>*` too?

**Review by byCedric — APPROVED** [review]
Looks great, one mention related to a docblock only

**Review by wesleytodd — APPROVED** [review]
Look great! This approach is basically the same as the "pass control back and forth instead of adding start checks in the caller" I as thinking would work well for the scope & breaking parts of the summary.

**Pull request merged**